File: src/main/java/name/gyger/jmoney/util/DateUtil.java
package name.gyger.jmoney.util;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class DateUtil {

    private static SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");

    public static Date parse(String dateString) {
        if (dateString == null) return null;

        Date result = null;
        try {
            result = sdf.parse(dateString);
        } catch (ParseException e) {
            throw new IllegalArgumentException(dateString, e);
        }
        return result;
    }

}

File: src/main/kotlin/name/gyger/jmoney/util/DateUtil.kt
package name.gyger.jmoney.util

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class DateUtil

private val sdf = SimpleDateFormat("yyyy-MM-dd")

fun parse(dateString: String?): Date? {
    if (dateString == null) return null

    val result: Date?
    try {
        result = sdf.parse(dateString)
    } catch (e: ParseException) {
        throw IllegalArgumentException(dateString, e)
    }

    return result
}

